torch/csrc/jit/codegen/cuda/lower_shift.cpp

```diff
@@ -801,7 +796,8 @@ std::unordered_map<IterDomain*, Val*> HaloInfo::buildConcreteHaloExtentMap(
         merged_shifted_ids.insert(ir_utils::caMapExactConcreteId(merge->out()));
         // Note that halo_width_map_ is not updated
       } else {
-        setHaloWidth(ir_utils::caMapExactConcreteId(merge->out()), 0);
+        local_halo_info.setHaloWidth(
+            ir_utils::caMapExactConcreteId(merge->out()), 0);
```

Comment:
Regarding the decoupling, here we still use the CA map held by GpuLower. And, also just realized it's a little counter-intuitive that `ir_utils` implicitly uses `GpuLower`. It doesn't seem trivial to remove this interface as it's used everywhere.

Reply:
Is making this `local_halo_info` right?

Reply:
Oh, this class is really hard set to the lowering process as it's taking indexing in. That's why I didn't bother looking more closely.

Reply:
`ir_utils::caMapExactConcreteId` seems like a weak alias, will get rid of that either before this is merged or in another PR.

Reply:
Creating `local_halo_info` is right. It was part of Shiming's indexing refactoring. Previously, reference domains are added to the singleton instance of `HaloInfo`, but as we don't create reference IterDomains anymore, the properties generated for reference IterDomains are instead generated for concrete IterDomains. A complexity about halo is that even exact mapped IterDomains may have different halo sizes, so it's not possible to associate a unique halo attribute with a group of mapped IterDomains. As the halo size information depends on the loop nest where an expression appears, what Shiming did was to clone the pre-built HaloInfo and modify the halo information for the used concrete IterDomains at that loop nest. 

That's my understanding from my reading of his refactoring PR. Tagging @shmsong as he may want to chime in.

Reply:
Yeah, I was assuming it was just a typo.